integration-tests/trait-dyn-cross-contract-calls/lib.rs

```diff
@@ -0,0 +1,110 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// We can't run E2E test without the contract. So add `Dummy` contract.
+#[ink::contract]
+pub mod dummy {
```

Comment:
Instead of having this `dummy` contract can we not have the `caller` contract be the central point of these tests?

It's the one that we're interacting with anyways. This would also simplify the test setup a little bit (one less crate)